include/aws/crt/mqtt/Mqtt5Client.h

```diff
@@ -365,6 +367,26 @@ namespace Aws
 
                 static void s_clientTerminationCompletion(void *complete_ctx);
 
+                /**
+                 * A handler to access and do operation on underlying native client.
+                 * params:
+                 *  - aws_mqtt5_client* client: native client to processing
+                 *  - void* user_data : user defined user data passed in by the caller
+                 *  - void* out_data :  user defined out data as out-parameter
+                 */
+                using processingNativeClientHandler =
+                    std::function<bool(aws_mqtt5_client *, Allocator *, void *, void **)>;
+
+                /*
+                 * This is an evil help function to allow accessing the native client. It will call the
+                 * callback with native client and user_data, then return the result of the callback
+                 */
+                bool ProceedOnNativeClient(
```

Comment:
The function is a workaround to access the native client protected by the CPP mqtt5Client. This should no longer a thing after the new clean shutdown progress.